### PR TITLE
More direct glob dumping for reduction `unfold`

### PR DESCRIPTION
**Pull request by SkySkimmer**
see also https://github.com/coq/coq/issues/7861

### CODE REVIEW

**Comment by SkySkimmer** [comment]
@coqbot bench

**Comment by SkySkimmer** [comment]
@coqbot run full ci

I don't remember why I wanted to bench this so I won't restart the bench

File: plugins/ltac/tacintern.ml [hunk 1]
```diff
@@ -318,6 +318,13 @@ let evalref_of_globref ?loc ?short = function
       Nametab.pr_global_env Id.Set.empty r ++ spc () ++
       str "into an evaluable reference.")
 
+let evalref_of_globref ?loc ?short r =
+  let () = match r with
+    | GlobRef.VarRef x when (try ignore (Global.lookup_named x); false with Not_found -> true) -> ()
```
Comment (ppedrot) [inline plugins/ltac/tacintern.ml:323]: I'm already not super fan of where clauses but this one is particularly nasty. Can you expand it into an if-then-else and optionally make this a little bit more understandable at first sight?

Comment (SkySkimmer) [reply plugins/ltac/tacintern.ml]: changed

**Review by ppedrot — APPROVED** [review]

**Comment by ppedrot** [comment]
@coqbot merge now

**Pull request merged**